### PR TITLE
Add S3 data source

### LINKED ISSUE #467: Document the S3 source
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/6846

### Which part(s) of the docs might be affected or should be updated? And how?

In the Data ingestion section, create a new topic 'S3' or 'Files' to describe the source. 

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info
- **Description**: 
Add S3 data source doc.
Edit typos in Pulsar doc.
Change "materialized view" syntax in CDC doc to "table" following new syntax change.

- **Preview**: 
[ Paste the preview link to the edited page here. ]

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/6846

- **Related doc issue**: 
https://github.com/risingwavelabs/risingwave-docs/issues/467

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

**Comment by emile-00** [comment]
@waruto210 Can you double-check if I have the correct parameters included and if there are any other ones I should add? Also, can you give a short description of what match_pattern is? Thank you!

File: docs/create-source/create-source-s3.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Use the SQL statement below to connect RisingWave to an Amazon S3 source.
+
+## Syntax
+
+```sql
+CREATE [ MATERIALIZED ] SOURCE [ IF NOT EXISTS ] source_name 
+[schema_definition]
+WITH (
+   connector='s3',
+   field_name='value', ...
```
Comment (waruto210) [inline docs/create-source/create-source-s3.md:16]: For s3 connector and csv format, we do not support specifying certain columns at this time, also, the order and type of the columns in `schema_definition` must be exactly the same as in the csv, so `schema_definition` is not optional but required here.

File: docs/create-source/create-source-s3.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+)
+ROW FORMAT data_format;
```
Comment (waruto210) [inline docs/create-source/create-source-s3.md]: s3 source can only be used with csv format now, the syntax should be like:
```
create SOURCE/TABLE [ IF NOT EXISTS ] source_name
schema_definition
with (
    connector = 's3',
    s3.region_name = 'ap-southeast-1',
    s3.bucket_name = 'xxx',
    s3.credentials.access = 'xxx',
    s3.credentials.secret = 'xxx'
) row format csv [WITHOUT HEADER] delimited by ',';
```

`WITHOUT HEADER` means the first line of csv files in the bucket is a header.

File: docs/create-source/create-source-s3.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+For materialized sources with primary key constraints, if a new data record with an existing key comes in, the new record will overwrite the existing record. 
+
+:::
+
+## Parameters
+
+|Field|Notes|
+|---|---|
+|`MATERIALIZED`| When you materialize a source, you choose to persist the data from the source in RisingWave.|
+|s3.service_name	|Required. The service region.|
```
Comment (waruto210) [inline docs/create-source/create-source-s3.md]: not `service_name`, but `region_name`.

File: docs/create-source/create-source-s3.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+|s3.bucket_name	|Required. The name of the bucket the data source is stored in.	|
+|match_pattern| Conditional.|
```
Comment (waruto210) [inline docs/create-source/create-source-s3.md]: match pattern is used to match some specific files under the bucket, currently not supported, will be implemented later, please remove it first.

File: docs/create-source/create-source-s3.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+|s3.credentials.access| Conditional. This field indicates the access key ID of AWS. It must appear in pairs with s3.credentials.secret.|
+|s3.credentials.secret| Conditional. This field indicates the secret access key of AWS. It must appear in pairs with s3.credentials.access.|
```
Comment (waruto210) [inline docs/create-source/create-source-s3.md]: In addition, if the user does not specify these two parameters, RW will try to use `~/.aws/credentials`.

File: docs/create-source/create-source-s3.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+```sql
+(
+   column_name data_type [ PRIMARY KEY ], ...
+   [ PRIMARY KEY ( column_name, ... ) ]
+)
+```
+
+:::info
+
+For Avro and Protobuf data, do not specify `schema_definition` in the `CREATE SOURCE` or `CREATE MATERIALIZED SOURCE` statement. The schema should be provided in a Web location in the `ROW SCHEMA LOCATION` section.
```
Comment (waruto210) [inline docs/create-source/create-source-s3.md]: s3 source can only be used with csv format now.

File: docs/create-source/create-source-s3.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+title: Ingest data from databases with S3 source.
+description: Ingest data from databases with S3 source.
+slug: /create-source-s3
+---
+Use the SQL statement below to connect RisingWave to an Amazon S3 source.
+
+## Syntax
+
+```sql
+CREATE [ MATERIALIZED ] SOURCE [ IF NOT EXISTS ] source_name
```
Comment (waruto210) [inline docs/create-source/create-source-s3.md]: We have deprecated `CREATE MATERIALIZED SOURCE` now.

**Review by waruto210 — APPROVED** [review]
LGTM.

File: docs/create-source/create-source-s3.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+   column_name data_type [ PRIMARY KEY ], ...
+   [ PRIMARY KEY ( column_name, ... ) ]
+)
+```
+
+## Parameters
+
+|Field|Notes|
+|---|---|
+|`MATERIALIZED`| When you materialize a source, you choose to persist the data from the source in RisingWave.|
```
Comment (hengm3467) [inline docs/create-source/create-source-s3.md]: As we have deprecated `create materialized source`, and replaced it with `create table with connector...` syntax. We do not need the `MATERIALIZED` row here.

File: docs/create-source/create-source-s3.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
+---
+id: create-source-s3
+title: Ingest data from databases with S3 source.
```
Comment (hengm3467) [inline docs/create-source/create-source-s3.md]: It's not accurate to describe it as loading data from databases. In this scenario, RisingWave is the target, and we can treat S3 buckets here as just files in CSV format. Let's think about if there are better ways to put this. Please see the descriptions on this page for reference: https://docs.imply.io/polaris/api-ingest-s3/

File: sidebars.js [hunk 10]
```diff
@@ -177,6 +177,11 @@ const sidebars = {
           label: 'Astra Streaming',
           id: 'guides/connector-astra-streaming',
         },
+        {
+          type: 'doc',
+          label: 'S3',
```
Comment (hengm3467) [inline sidebars.js]: S3 -> Amazon S3

**Review by tabVersion — APPROVED** [review]

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**